drivers/adc/ad4858/ad4858.c

```diff
@@ -685,46 +707,83 @@ int ad4858_spi_data_read(struct ad4858_dev *dev, struct ad4858_conv_data *data)
 	if (ret)
 		return ret;
 
-	switch (dev->packet_format) {
-	case AD4858_PACKET_20_BIT:
-		/* 20-bit conversion result */
-		for (chn = 0; chn < AD4858_NUM_CHANNELS; chn++) {
-			indx = buff_chn_offset[chn];
-			if (!(chn % 2))
+	switch (dev->prod_id) {
+	case AD4855_PROD_ID_L:
+		switch (dev->packet_format) {
+		case AD4858_PACKET_16_BIT:
+			/* 16-bit conversion result */
+			for (chn = 0; chn < AD4858_NUM_CHANNELS; chn++) {
+				indx = chn * 2;
+				data->raw[chn] = ((uint16_t)buff[indx] << 8) |
+						 ((uint16_t)buff[indx + 1]);
+			}
+			break;
+
+		case AD4858_PACKET_24_BIT:
+			/* 16-bit conversion result + 1-bit OR/UR + 3-bit channel ID + 4-bit softspan ID */
```

Comment:
I would love to see more of NO_OS_BIT, NO_OS_GENMASK, no_os_field_prep, no_os_field_get being used. Some of these get pretty complex and hard to read. I think proper mask macros with proper names would help here. This comment applies to all the modifications not just this line.